types/calidation/index.d.ts

```diff
@@ -1,134 +1,159 @@
-// Type definitions for calidation 1.16
+// Type definitions for calidation 1.17
 // Project: https://github.com/selbekk/calidation#readme
 // Definitions by: Ray Knight <https://github.com/ArrayKnight>
+//                 Lisa Tassone <https://github.com/lisatassone>
 // Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
 // TypeScript Version: 3.5
 
 import * as React from 'react';
 
-export interface Dictionary<T = any> {
-    [key: string]: T;
-}
+export type Dirty<T extends object = any> = Record<keyof T, boolean>;
+
+export type Errors<T extends object = any> = Record<keyof T, string | null>;
 
-export type Dirty = Dictionary<boolean>;
+export type Fields<T extends object = any> = T;
 
-export type Errors = Dictionary<string | null>;
+export type Transform<T extends object = any> = Record<keyof T, (value: any) => any>;
```

Comment:
I think the transforms should accept any as the value but return the type of the field.

Reply:
Done. I've removed the additional transforms func def since it probably isn't capturing the intent of that feature.